sync/sync.go

```diff
@@ -373,3 +373,58 @@ func (s *StringMap) SetString(val string) error {
 	s.Set(dict)
 	return nil
 }
+
+// StringSlice is a []string type with concurrent access support.
+type StringSlice struct {
+	rw    sync.RWMutex
+	value []string
+}
+
+// Get returns the internal value.
+func (s *StringSlice) Get() []string {
+	s.rw.RLock()
+	defer s.rw.RUnlock()
+	return s.value
+}
+
+// Set a value.
+func (s *StringSlice) Set(value []string) {
+	s.rw.Lock()
+	defer s.rw.Unlock()
+	s.value = value
+}
+
+// MarshalJSON returns the JSON encoding of the value.
+func (s *StringSlice) MarshalJSON() ([]byte, error) {
+	s.rw.RLock()
+	defer s.rw.RUnlock()
+	return json.Marshal(s.value)
+}
+
+// UnmarshalJSON returns the JSON encoding of the value.
+func (s *StringSlice) UnmarshalJSON(d []byte) error {
+	s.rw.RLock()
+	defer s.rw.RUnlock()
+	return json.Unmarshal(d, &s.value)
+}
+
+// String returns a string representation of the value.
+func (s *StringSlice) String() string {
+	s.rw.RLock()
+	defer s.rw.RUnlock()
+	return strings.Join(s.value, ",")
+}
+
+// SetString parses and sets a value from string type.
+func (s *StringSlice) SetString(val string) error {
+	slice := make([]string, 0)
+	if val == "" || strings.TrimSpace(val) == "" {
+		s.Set(slice)
+		return nil
+	}
+	for _, item := range strings.Split(val, ",") {
+		slice = append(slice, strings.TrimSpace(item))
+	}
+	s.Set(slice)
```

Comment:
Q: Do we want to trim spaces? 
I'm not aware of any use cases, just thinking about it since trimming trailing spaces seems more strict than not trimming them

Reply:
The reason is that this will often be used for configuration, and a leading/trailing space can be hard to spot when something is failing.
The original `stringMap` was first code in API Gateway, and we actually had an issue during deployment because of a space, which is why this code has trims :)

Reply:
Makes sense 👍 . 
I guess we can assume that in most cases we don't need leading and trailing spaces.

And as discussed, **if we need** more advanced options, such as supporting leading/trailing spaces and `,` we could introduce a more powerful mechanism (such as quotes and escaping).